problem_builder/tasks.py

```diff
@@ -194,32 +197,24 @@ def _get_submissions(course_key_str, block, user_id):
         return sub_api.get_submissions(student_dict, limit=1)
 
 
-def _get_user_info(users, submission):
-    """
-    Return a (username, user id, user email) tuple for the user who provided `submission`.
-
-    If the anonymous ID of the submission can't be resolved into a user,
-    (student ID, 'N/A', 'N/A') is returned
-    """
-    user = None
-    if users is not None:
-        user = next(user for user in users if user.anonymous_user_id == submission['student_id'])
-
-    if user is None:
-        return (submission['student_id'], 'N/A', 'N/A')
-    return (user.username, user.id, user.email)
-
-
 def get_users_by_anonymous_ids(anonymous_ids):
     """
     Return users by anonymous_ids using AnonymousUserId lookup table.
     """
     if not anonymous_ids:
         return None
 
-    return User.objects.filter(
+    users = User.objects.filter(
         anonymoususerid__anonymous_user_id__in=anonymous_ids
-    ).annotate(anonymous_user_id=F('anonymoususerid__anonymous_user_id')).iterator()
+    ).annotate(
+        anonymous_user_id=F('anonymoususerid__anonymous_user_id')
+    ).values(
+        'anonymous_user_id', 'username', 'id', 'email'
+    ).iterator()
+
+    return {
+        user['anonymous_user_id']: (user['id'], user['username'], user['email']) for user in users
```

Comment:
The tuple order here is not the same as the one returned by _get_user_info, which means that the reports contain the user id in place of the username.

Reply:
@xitij2000, thanks for catching this. I have fixed this now.